UI/MapWnd.cpp

```diff
@@ -6985,9 +6988,51 @@ void MapWnd::DispatchFleetsExploring() {
                 auto pair = GetPathfinder()->ShortestPath(
                     fleet->SystemID(), system_supply.first, empire_id);
 
+                auto route = pair.first;
+
+                // check if path contains hostile ships/fleets
+                bool is_unobstructed = true;
+                if (!ignore_hostile_route) {
+                    // TODO: replace with a general purpose universe, empire, or pathfinder function
+                    for (auto next_system = ++(route.begin()); next_system != route.end(); ++next_system) {
+                        for (auto object_id : GetSystem(*next_system)->ContainedObjectIDs()) {
+                            auto object = GetUniverseObject(object_id);
+                            if (!object) {
+                                ErrorLogger() << "Unable to find object with id: " << object_id;
+                                continue;
+                            }
+
+                            // Only ships/fleets are of concern to exploring ships
+                            auto object_type = object->ObjectType();
+                            if (object_type != OBJ_SHIP && object_type != OBJ_FLEET)
+                                continue;
+
+                            // Owned by non-hostile empire
+                            if (object->Owner() != ALL_EMPIRES &&
+                                Empires().GetDiplomaticStatus(empire_id, object->Owner()) != DIPLO_WAR)
+                            { continue; }
+
+                            auto object_vis_map = GetUniverse().GetObjectVisibilityTurnMapByEmpire(object_id, empire_id);
```

Comment:
What's the purpose of this check?

Reply:
Checks if empire has observed this object, added comment to clarify

Reply:
Since this is running on the client, when would an empire's client know about an object to run that check if it hadn't observed it?

Reply:
TY, I did not realize they were limited to client info.  I've removed this check.